src/azul/drs.py

```diff
@@ -97,10 +192,17 @@ def _uri_to_url(self, drs_uri: str, access_id: Optional[str] = None) -> str:
         compact, identifier-based URIs (drs://[provider_code/]namespace:accession)
         are not.
         """
-        parsed = furl(drs_uri)
+        parsed = DRSCompactPseudoURI.maybe_parse(drs_uri)
         scheme = 'drs'
         require(parsed.scheme == scheme,
                 f'The URI {drs_uri!r} does not have the {scheme!r} scheme')
+        if isinstance(parsed, DRSCompactPseudoURI):
+            object_id = parsed.object_id
+            parsed = self.id_client.prefix_to_host(parsed.prefix)
+        else:
+            object_id = one(parsed.path.segments)
+            parsed.scheme = 'https'
+        assert isinstance(parsed, furl)
```

Comment:
With a class hierarchy approach, this branch would be dissolved into an abstract method with two implementations.